src/sys/unix.rs

```diff
@@ -538,6 +570,35 @@ impl Socket {
         Ok((n as usize, addr))
     }
 
+    #[cfg(not(target_os = "redox"))]
+    pub fn recv_vectored(&self, bufs: &mut [IoSliceMut<'_>], flags: c_int) -> io::Result<usize> {
+        let mut msg: libc::msghdr = unsafe { std::mem::zeroed() };
```

Comment:
Nit: to avoid the `unsafe` block I think you can just create `libc::msghdr` properly.

Reply:
Done :)